### ISSUE TITLE
Listbox reactivity breaks with undefined

### ISSUE THREAD

**Issue by atyshka** [tailwindlabs/headlessui#2622]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

1.7.13

**What browser are you using?**

Chrome & Safari

**Reproduction URL**

https://codesandbox.io/p/sandbox/gracious-dhawan-p8dnwy?file=%2Fsrc%2FApp.vue%3A71%2C2

The listbox model resets to undefined every 2s. Select an option from the listbox, and notice that it still appears selected in the options menu even after the listbox value resets back to undefined.

**Describe your issue**

When setting a Listbox's modelValue to undefined, the internal state's .value does not update. Then the listbox options menu retains the last set value. Something is broken with reactivity linking the modelValue prop to the Listbox's .value state.

**Comment by RobinMalfait** [comment]
Hey, thank you for this bug report! 🙏

The TL;DR is to use `null` instead of `undefined`.

---

This is happening because the `Listbox` can be used as a controlled and as an uncontrolled component. In case you don't know the difference: A controlled component is where you control the value from the outside and pass it in, an uncontrolled component manages its state internally inside the component.

We do need to differentiate between those 2 cases, so we consider the `Listbox` in a "controlled" state if the incoming value is set to _something_ or `null`. It is considered "uncontrolled" if the value is `undefined`. This is what's happening in your case as well, you are making it "uncontrolled" which means that the state is handled inside the component.

In your case using `null` instead of `undefined` should solve that issue.

One reason why we can't use `undefined` is because in JS passing `undefined` or not passing anything would result in `undefined`:

```js
function test(value) {
  console.log({value})
}
test()
test(undefined)
``` 

<img width="269" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/9fc47439-aa38-4f83-ad10-8c9057f387c5">


Hope this helps!
